Export poll results

Senen opened the pull request:
## References

Ported from:
* rockandror/innoviris-consul#35
* rockandror/innoviris-consul#36
* rockandror/innoviris-consul#40

## Objectives
Add the "Download poll results" link to download a CSV file with all answers.

## Visual Changes
![Link to download a csv file in admin polls show page](https://user-images.githubusercontent.com/16189/174019680-391dd741-27c3-482a-8ac2-44aa60257484.png)

## Notes

The format I have chosen to display all the answers is as follows:
| Question 1 Title | Question 2 Title | Question 3 Title | ... | Question N Title|
| ------------- | ------------- | ------------- | ------------- | ------------- |
| Answer for question 1 | Answer for question 2 | Answer for question 3 | ... | Answer for question N |
| Answer for question 1 | Answer for question 2 | Answer for question 3 | ... | Answer for question N |

taitus committed:
Copy original poll admin show view to custom folder

taitus committed:
Allow export answers from polls

Add new Class Exporter to allow donwload csv file with all answers from
polls.
The link to download the csv file is in admin poll edit page.

taitus committed:
Allow exporting answers without reply

javierm reviewed (commented):
IMHO we need to add an extra change here. It shouldn't be possible to download the CSV file (that is, no link to the URL, and an unauthorized error when accessing the URL manually) until the poll has finished :thinking:. Otherwise it'd be possible for administrators to know the results while the poll is still open.

Senen committed:
Use a different action to export polls results and allow administrators to do it

The admin poll show action should be always accesible no matter the status of
the poll. However the export feature should only be availabe once the poll is
done, just like the poll results.

As the request format is not available in the abilities context we would need to
change the abilities implementation to pass in the format. In place of doing that,
I'm creating a new action to export the poll results within de admin poll results
controller which has more sense to me, so now we can fully control permissions
without needing the request format.

Senen committed:
Allow to export poll results once the poll ends
